onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -162,6 +162,47 @@ void convTransposeWithDynamicPadsShapeInference(InferenceContext& ctx) {
     return;
   }
 }
+
+void embedLayerNormalizationShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 2, 0);
+  propagateElemTypeFromInputToOutput(ctx, 0, 1);
+  if (!hasInputShape(ctx, 0))
+    return;
+
+  auto& input_ids_shape = getInputShape(ctx, 0);
+  auto& input_ids_dims = input_ids_shape.dim();
+
+  // Note that both batch size and sequence length could be symbolic.
+  // So we only check dimension size here.
+  if (input_ids_dims.size() != 2) {
+    fail_shape_inference("Inputs 0 shall be 2 dimensions");
+  }
+
+  // get hidden_size from the last dimension of embedding
```

Comment:
Something should check that the other embeddings/layernorm params have the same `hidden_size`, and the position embedding has the expected sequence length. We at least need a runtime check, but putting it in the shape inference function is nice-to-have as well.

Reply:
Apologies, I completely missed that!